### PR TITLE
Validate natives in composite types

### PR DESCRIPTION
**Pull request by pimeys**
Closes: https://github.com/prisma/prisma/issues/12460

### CODE REVIEW

**Review by tomhoule — COMMENTED** [review]
We want to reevaluate native types on mongo this cycle (should we even have them?), so all of this is likely to change.

**Comment by pimeys** [comment]
The work is additional on this PR. We should anyhow run the validations.

File: libs/datamodel/parser-database/src/walkers/composite_type.rs [hunk 1]
```diff
@@ -118,6 +118,19 @@ impl<'db> CompositeTypeFieldWalker<'db> {
...
     }
 
+    /// The type of the field in case it is a scalar type (not an enum, not a composite type).
+    pub fn scalar_type(self) -> Option<ScalarType> {
+        let mut r#type = self.r#type();
+
+        loop {
+            match r#type {
+                ScalarFieldType::BuiltInScalar(scalar) => return Some(*scalar),
+                ScalarFieldType::Alias(alias_id) => r#type = &self.db.types.type_aliases[alias_id],
```
Comment (do4gr) [inline libs/datamodel/parser-database/src/walkers/composite_type.rs:128]: We already catch loops somewhere else?

Comment (pimeys) [reply libs/datamodel/parser-database/src/walkers/composite_type.rs:128]: This one just resolves the actual type. Copied from the `scalar_type.rs`.

Comment (pimeys) [reply libs/datamodel/parser-database/src/walkers/composite_type.rs:128]: Loops are catched way earlier.

**Pull request merged**